File: src/main/java/com/arsene/eportfolio/model/entities/Project.java
package com.arsene.eportfolio.model.entities;

import lombok.Data;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Entity
@Data
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    private String icon;
    private String description;
    private String git;
    private String color;


    @ManyToMany
    private Set<Technology> technologies;

    @ManyToMany(fetch = FetchType.EAGER)
    private Set<Ability> abilities;

    @ManyToOne
    private Subject subject;


}

File: frontend/src/app/core/model/Project.ts
import {Ability} from "./Ability";
import {Technology} from "./Technology";

export class Project {
  constructor(
    public id?: number,
    public name?: string,
    public icon?: string,
    public description?: string,
    public git?: string,
    public color?: string,
    public abilities?: Array<Ability>,
    public technologies?: Array<Technology>
  ) {
    if(!abilities){
      this.abilities = new Array<Ability>();
    }
    if(!technologies){
      this.technologies = new Array<Technology>();
    }
  }
}
